airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.scheduler;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import io.airbyte.analytics.TrackingClient;
+import io.airbyte.analytics.TrackingClientSingleton;
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.commons.map.MoreMaps;
+import io.airbyte.config.JobConfig.ConfigType;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.StandardSyncSchedule;
+import io.airbyte.config.helpers.ScheduleHelpers;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class JobTracker {
+
+  public enum JobState {
+    STARTED,
+    SUCCEEDED,
+    FAILED
+  }
+
+  public static final String MESSAGE_NAME = "Connector Jobs";
+
+  private final ConfigRepository configRepository;
+  private final TrackingClient trackingClient;
+
+  public JobTracker(ConfigRepository configRepository) {
+    this(configRepository, TrackingClientSingleton.get());
+  }
+
+  @VisibleForTesting
+  JobTracker(ConfigRepository configRepository, TrackingClient trackingClient) {
+    this.configRepository = configRepository;
+    this.trackingClient = trackingClient;
+  }
+
+  public void trackCheckConnectionSource(UUID jobId, UUID sourceDefinitionId, JobState jobState) {
+    Exceptions.swallow(() -> {
```

Comment:
What's the reason for swallowing all these?

Reply:
we never want a user to fail at what they are doing because our analytics tracking threw an exception.